assignment4/disjointset.py

```diff
@@ -0,0 +1,37 @@
+class DisjointSet:
+    """Implementation of a Disjoint Set. Takes integers as keys."""
```

Comment:
I think you find your implementation more obvious than it actually is :)
I think it would be helpful if you added some comments about some of the functions.